File: solutions/0899-binary-gap/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var binaryGap = function(n) {
    let gap = 0;
    let max = 0;
    n = n.toString(2);
    
    for(let i = 0; i < n.length; i++){        
        const bit = n[i];
        if(bit === "1"){
            if(gap > max){
                max = gap;
            }
            gap = 1;
        }else{
            gap += 1;
        }
    }
    
    
    return max;
};

File: solutions/0899-binary-gap/solution.py
class Solution:
    def binaryGap(self, n: int) -> int:
        best = 0
        dist = 0
        for c in bin(n)[3:]:
            dist += 1
            if c == "1":
                best = max(best, dist)
                dist = 0

        return best
